src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -76,6 +77,23 @@ public TracerSettings(IConfigurationSource source)
                                false;
 
             Integrations = new IntegrationSettingsCollection(source);
+
+            if (source == null)
+            {
+                GlobalTags = new Dictionary<string, string>();
+            }
+            else
+            {
+                var gTags = source.GetString(ConfigurationKeys.GlobalTags);
```

Comment:
Code style suggestion: We can simplify the nested `if` statements here by combining the check for null `source`:
```suggestion
                var gTags = source?.GetString(ConfigurationKeys.GlobalTags);
```
Then we don't need the `if (source == null)` above.

Reply:
👍 I originally had a more complex set of logic which is why I didn't originally use `?.` but now makes sense.